lib/fs.js

```diff
@@ -1097,7 +1097,9 @@ function writeAll(fd, buffer, offset, length, position, callback) {
       } else {
         offset += written;
         length -= written;
-        position += written;
+        if (position !== null) {
```

Comment:
To elaborate, `position` is internal to `writeFileSync` and `writeFile`. In the latter case via an internal `writeAll` function that's only called from `writeFile`.